pkg/monitor/datapath_drop.go

```diff
@@ -99,11 +99,14 @@ func (n *DropNotify) decodeDropNotify(data []byte) error {
 	n.Ifindex = byteorder.Native.Uint32(data[32:36])
 
 	switch n.Version {
+	case DropNotifyVersion0, DropNotifyVersion1:
 	case DropNotifyVersion2:
 		if l := len(data); l < dropNotifyV2Len {
-			return fmt.Errorf("unexpected DropNotify data length, expected %d but got %d", dropNotifyV2Len, l)
+			return fmt.Errorf("unexpected DropNotifyV2 data length, expected %d but got %d", dropNotifyV2Len, l)
 		}
 		n.Flags = data[36]
+	default:
+		return fmt.Errorf("Unrecognized drop event (version %d)", n.Version)
 	}
```

Comment:
Not necessarily to be addressed in this PR, but I wonder whether `switch` is the best control flow for this logic as we will want the v2 codepath to be executed when we introduce v3. I think something like:

```go
if n.Version >= DropNotifyVersion2 {
	// v2 codepath
}

// max known version check
if n.Version > DropNotifyVersion2 {
	// unknown version codepath
}
```

would make it easier to foresee how v3 would be introduced:

```go
if n.Version >= DropNotifyVersion2 {
	// v2 codepath
}
if n.Version >= DropNotifyVersion3 {
	// v3 codepath
}

// max known version check
if n.Version > DropNotifyVersion3 {
	// unknown version codepath
}
```

What do you think?

Reply:
Yep, I was thinking of adding new cases on top of the switch and then `fallthrough`, but this looks better, ty 👍🏼

Reply:
Ah yes the check order in the patch make sense as we check for length before all anything else :+1: